pkgs/tools/misc/nncp/default.nix

```diff
@@ -9,24 +10,29 @@
 
 stdenv.mkDerivation rec {
   pname = "nncp";
-  version = "5.3.3";
+  version = "5.4.1";
 
   src = fetchurl {
     url = "http://www.nncpgo.org/download/${pname}-${version}.tar.xz";
-    sha256 = "1l35ndzrvpfim29jn1p0bwmc8w892z44nsrdnay28k229r9dhz3h";
+    sha256 = "0qzs4zkaj6qmg7ghqab4ba07p7w5qkw05q8pbh8zhgaka6dhqbd0";
   };
 
-  nativeBuildInputs = [ go ];
+  nativeBuildInputs = [ go redo-apenwarr ];
 
-  preConfigure = ''
+  buildPhase = ''
     export GOCACHE=$PWD/.cache
+    export CFGPATH=/etc/nncp.hjson
+    export SENDMAIL=/run/wrappers/bin/sendmail # just a default value for generated config file
```

Comment:
This won't work on NixOS and adding defaults which are broken is not great.

Reply:
see https://github.com/NixOS/nixpkgs/pull/99304#discussion_r530870941

Reply:
I changed it to plain "sendmail", which is portable. And sending mail is optional and not a dependency. The generated config file is to be edited by the user anyway.